### ISSUE TITLE
[REVIEW]: BluVision Macro - a software for automated powdery mildew and rust disease quantification on detached leaves.

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2259]
**Submitting author:** @snowformatics (<a href="http://orcid.org/0000-0003-0536-835X">Stefanie Lueck</a>)
**Repository:** <a href="https://github.com/snowformatics/macrobot" target ="_blank">https://github.com/snowformatics/macrobot</a>
**Version:** 0.4.6
**Editor:** @mjsottile
**Reviewer:** @SirSharpest, @phckopper, @jizecn
**Archive:** <a href="https://doi.org/10.5281/zenodo.3929095" target="_blank">10.5281/zenodo.3929095</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/5aeae9ae09407cdf0ed1801c5f9b4fd7/status.svg)](https://joss.theoj.org/papers/5aeae9ae09407cdf0ed1801c5f9b4fd7)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/5aeae9ae09407cdf0ed1801c5f9b4fd7"><img src="https://joss.theoj.org/papers/5aeae9ae09407cdf0ed1801c5f9b4fd7/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/5aeae9ae09407cdf0ed1801c5f9b4fd7/status.svg)](https://joss.theoj.org/papers/5aeae9ae09407cdf0ed1801c5f9b4fd7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/snowformatics/macrobot" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@SirSharpest & @phckopper & @jizecn, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @mjsottile know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/snowformatics/macrobot">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@snowformatics) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [X] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [X] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [X] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/snowformatics/macrobot">repository url</a>?
- [X] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@snowformatics) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [X] **Installation:** Does installation proceed as outlined in the documentation?
- [X] **Functionality:** Have the functional claims of the software been confirmed?
- [X] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [X] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [X] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [X] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [X] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [X] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [X] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [X] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [X] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/snowformatics/macrobot">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@snowformatics) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @SirSharpest, @phckopper, @jizecn it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1101/2020.03.16.993451 is OK
- 10.7717/peerj.453 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1016/j.neuroscience.2009.01.027 is OK
- 10.1186/s13059-014-0518-8 is OK
- 10.1080/07060660509507230 is OK
- 10.1038/sj.bjp.0703793 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02259/joss.02259/10.21105.joss.02259.pdf)

**Comment by mjsottile** [comment]
:wave: @phckopper, @jizecn - I just wanted to check in and see if you were making progress on the review. I see the activity in the linked comment above for @SirSharpest review, so that seems to be proceeding well.  Feel free to let me know if you need any assistance in conducting the review.

**Comment by phckopper** [comment]
Hello, just checking in and saying that I'll be making my review this week.

**Comment by jizecn** [comment]
>  @phckopper, @jizecn - I just wanted to check in and see if you were making progress on the review. I see the activity in the linked comment above for @SirSharpest review, so that seems to be proceeding well. Feel free to let me know if you need any assistance in conducting the review.

@mjsottile Hi, I was trying to review the work, but could not click the checklist buttons above. 
I got the error of _"Sorry, we couldn't find that repository invitation. It is possible that the invitation was revoked or that you are not logged into the invited account."_. 
Can you check please? Thanks.

**Comment by danielskatz** [comment]
@whedon re-invite @jizecn as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@jizecn please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by snowformatics** [comment]
Dear reviewers @phckopper, @jizecn, @SirSharpest 

I uploaded an image test datasets which could be downloaded manually and used with the software. Please unzip the file and use the data folder as main entry for the software, e.g. 
`-s C:\Users\name\Desktop\data\ -d C:\Users\name\Desktop\out\ -p mildew`

The folder structure should look like this (it's a bit complex because we use the software in batch mode):
https://github.com/snowformatics/macrobot/blob/master/docs/images/folder_structure.png

As jizecn pointed out, Python Version should be 3.8. I will update the documentation.

I hope it will be easier to test the software now. Please let me know if you have any problems.

Best regards
Stefanie

**Comment by phckopper** [comment]
My review is now done. Nice project!

**Comment by snowformatics** [comment]
@whedon set 10.5281/zenodo.3929095 as archive

**Comment by whedon** [comment]
I'm sorry @snowformatics, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by mjsottile** [comment]
@snowformatics I can set the archive once the review is complete.  It looks like we are just waiting to finish the issue where @SirSharpest is conducting their part of the review.  Once that is done I’ll set the DOI and work through the last steps of the review.

**Comment by snowformatics** [comment]
> @snowformatics I can set the archive once the review is complete. It looks like we are just waiting to finish the issue where @SirSharpest is conducting their part of the review. Once that is done I’ll set the DOI and work through the last steps of the review.

Sorry for the command, I just wanted to  note down the link, no rush...

**Comment by mjsottile** [comment]
No problem: thanks for providing the link.  That will speed things up for me when we’re ready to finish.

**Comment by SirSharpest** [comment]
@mjsottile My review is complete too. I can recommend for publication. 

(My one minor gripe, which I can understand letting go, is software tests should run out of the box without a user needing to source additional data)

**Comment by jizecn** [comment]
@mjsottile Done from my side too. I would recommend for acceptance too. Good work.

**Comment by snowformatics** [comment]
Thanks everybody for taking the time to review our project!

@SirSharpest I will reconsider my tests for future projects in order to exclude integration tests with real world images. Thanks for the suggestion.

**Comment by mjsottile** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02259/joss.02259/10.21105.joss.02259.pdf)

**Comment by mjsottile** [comment]
It looks like the reviews are complete and we can proceed to final steps.  I did a pass over the paper and found a few minor issues that should take very little time to address.  They are listed in the project issue https://github.com/snowformatics/macrobot/issues/12.

Once you've addressed those I will submit the paper for final processing by the EIC team.

**Comment by mjsottile** [comment]
@whedon set 10.5281/zenodo.3929095 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3929095" target="_blank">10.5281/zenodo.3929095</a> is the archive.

**Comment by snowformatics** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02259/joss.02259/10.21105.joss.02259.pdf)

**Comment by mjsottile** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1101/2020.03.16.993451 is OK
- 10.1080/13102818.2017.1400401 is OK
- 10.1007/978-94-017-7752-0 is OK
- 10.7717/peerj.453 is OK
- 10.1186/s13059-014-0518-8 is OK
- 10.1007/s13593-012-0105-x is OK
- 10.1111/j.1364-3703.2011.00783.x is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1086/659307 is OK
- 10.1007/s10681-011-0360-y is OK
- 10.1016/j.neuroscience.2009.01.027 is OK
- 10.1007/s10658-009-9468-5 is OK
- 10.1080/07060660509507230 is OK
- 10.1038/sj.bjp.0703793 is OK

MISSING DOIs

- None

INVALID DOIs

- https://doi.org/10.1016/j.ecolecon.2011.03.016 is INVALID because of 'https://doi.org/' prefix
```

**Comment by mjsottile** [comment]
@snowformatics Looks like one final issue with one of the DOIs.  Can you fix that so I can issue the accept command?  I didn’t notice that when I did my final review of the paper earlier.

**Comment by mjsottile** [comment]
Thanks for your work on the review process @SirSharpest, @phckopper, @jizecn !

**Comment by snowformatics** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1101/2020.03.16.993451 is OK
- 10.1080/13102818.2017.1400401 is OK
- 10.1007/978-94-017-7752-0 is OK
- 10.7717/peerj.453 is OK
- 10.1186/s13059-014-0518-8 is OK
- 10.1007/s13593-012-0105-x is OK
- 10.1111/j.1364-3703.2011.00783.x is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1016/j.ecolecon.2011.03.016 is OK
- 10.1086/659307 is OK
- 10.1007/s10681-011-0360-y is OK
- 10.1016/j.neuroscience.2009.01.027 is OK
- 10.1007/s10658-009-9468-5 is OK
- 10.1080/07060660509507230 is OK
- 10.1038/sj.bjp.0703793 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by mjsottile** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1101/2020.03.16.993451 is OK
- 10.1080/13102818.2017.1400401 is OK
- 10.1007/978-94-017-7752-0 is OK
- 10.7717/peerj.453 is OK
- 10.1186/s13059-014-0518-8 is OK
- 10.1007/s13593-012-0105-x is OK
- 10.1111/j.1364-3703.2011.00783.x is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1016/j.ecolecon.2011.03.016 is OK
- 10.1086/659307 is OK
- 10.1007/s10681-011-0360-y is OK
- 10.1016/j.neuroscience.2009.01.027 is OK
- 10.1007/s10658-009-9468-5 is OK
- 10.1080/07060660509507230 is OK
- 10.1038/sj.bjp.0703793 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1547

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1547, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
Hi @snowformatics, just a few minor edits remaining on the paper before we accept:

- In the first paragraph (and anywhere else this occurs), when you have multiple citations please combine them like this:
`[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"
- the Hazell reference is missing details
- please add the DOI 10.1007/s10681-011-0361-x to the Huerta-Espino article
- a few references include "journal article" in the citations, please remove that where it appears

**Comment by snowformatics** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1101/2020.03.16.993451 is OK
- 10.1080/13102818.2017.1400401 is OK
- 10.1007/978-94-017-7752-0 is OK
- 10.7717/peerj.453 is OK
- 10.1186/s13059-014-0518-8 is OK
- 10.1007/s13593-012-0105-x is OK
- 10.1111/j.1364-3703.2011.00783.x is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1007/s10681-011-0361-x is OK
- 10.1016/j.ecolecon.2011.03.016 is OK
- 10.1086/659307 is OK
- 10.1007/s10681-011-0360-y is OK
- 10.1016/j.neuroscience.2009.01.027 is OK
- 10.1007/s10658-009-9468-5 is OK
- 10.1080/07060660509507230 is OK
- 10.1038/sj.bjp.0703793 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by snowformatics** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02259/joss.02259/10.21105.joss.02259.pdf)

**Comment by snowformatics** [comment]
> Hi @snowformatics, just a few minor edits remaining on the paper before we accept:
> 
> * In the first paragraph (and anywhere else this occurs), when you have multiple citations please combine them like this:
>   `[@author1:2001; @author2:2001]` -> "(Author1 et al., 2001; Author2 et al., 2002)"
> * the Hazell reference is missing details
> * please add the DOI 10.1007/s10681-011-0361-x to the Huerta-Espino article
> * a few references include "journal article" in the citations, please remove that where it appears

Thanks for your comment, I hope everything is ok now.

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1101/2020.03.16.993451 is OK
- 10.1080/13102818.2017.1400401 is OK
- 10.1007/978-94-017-7752-0 is OK
- 10.7717/peerj.453 is OK
- 10.1186/s13059-014-0518-8 is OK
- 10.1007/s13593-012-0105-x is OK
- 10.1111/j.1364-3703.2011.00783.x is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1007/s10681-011-0361-x is OK
- 10.1016/j.ecolecon.2011.03.016 is OK
- 10.1086/659307 is OK
- 10.1007/s10681-011-0360-y is OK
- 10.1016/j.neuroscience.2009.01.027 is OK
- 10.1007/s10658-009-9468-5 is OK
- 10.1080/07060660509507230 is OK
- 10.1038/sj.bjp.0703793 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1553

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1553, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1281278475467358208) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1554
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02259](https://doi.org/10.21105/joss.02259)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congrats @snowformatics on your article's publication in JOSS!

Many thanks to @SirSharpest, @phckopper, and @jizecn for reviewing this, and @mjsottile for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02259/status.svg)](https://doi.org/10.21105/joss.02259)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02259">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02259/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02259/status.svg
   :target: https://doi.org/10.21105/joss.02259
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02259/status.svg)](https://doi.org/10.21105/joss.02259)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
